XenModel/PoolJoinRules.cs

```diff
@@ -388,6 +397,35 @@ private static bool DifferentServerVersion(Host slave, Host master)
                 slave.ProductBrand != master.ProductBrand;
         }
 
+        /// <summary>
+        /// Check whether all updates that request homogeneity are in fact homogeneous
+        /// across the proposed pool. This is used in CanJoinPool and prevents the pool from being created
+        /// </summary>
+        private static bool DifferentHomogeneousUpdates(Host slave, Host master)

```

Comment:
I think it would be beneficial to check for server versions as all the below doesn't apply to pre-Ely hosts. Doing so would document the code.